src/QsCompiler/CommandLineTool/Options.cs

```diff
@@ -64,6 +64,20 @@ public class CompilationOptions : Options
             HelpText = "Additional properties to populate the AssemblyConstants dictionary with. Each item is expected to be of the form \"key:value\".")]
         public IEnumerable<string>? AdditionalAssemblyProperties { get; set; }
 
+        [Option(
+            "skip-monomorphization",
+            Required = false,
+            Default = false,
+            HelpText = "Specifies whether to omit monomorphizing the AST for executables.")]
+        public bool SkipMonomorphization { get; set; }
+
+        [Option(
+            "force-rewrite-step-execution",
+            Required = false,
+            Default = false,
+            HelpText = "Specifies whether to execute rewrite steps even if their precondition is not satisfied. If this is the case, the transformation output will be ignored and the compilation fails.")]
```

Comment:
I'm a little confused by this option, since it states that the compilation will fail. Why have an option that will always cause failure? I think I may be misunderstanding the comment. Either way, it looks like it is only used in a test so far, so perhaps mentioning in the help text that it is for testing purposes only would be helpful (assuming that is true)?

Reply:
Copy pasting the description of the original PR (https://github.com/microsoft/qsharp-compiler/pull/907):

The current behavior is that if the precondition verification fails for rewrite steps, then the step is skipped and the compilation succeeds with a warning. While overall this seems reasonable behavior given that rewrite steps may come from a 3rd party, I also found this rather suboptimal for testing: When I set up the test for qir emission for a targeted executable, the test could technically succeed when/because the targeting step is skipped (since we test for a specific .ll it will fail regardless, but still seems risky since it is easy to overlook that the transformation in question is not executed when e.g. updating the test).
I hence went ahead and added a command line flag to force execution of rewrite steps and generate an error when the precondition is not met (see the doc comments for more detail).

Reply:
Ah, that makes perfect sense then. I think what confused me was the second sentence, "If this is the case, the transformation output will be ignored and the compilation fails." I thought that the "case" was the parameter being specified, meaning every time the parameter is specified the output would be ignored and compilation fails. Consider this a minor feedback on the phrasing, but the intent definitely makes sense now.

Reply:
I updated the phrasing to make it clearer.